iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/util/RegulationUtils.java

```diff
@@ -3,15 +3,60 @@
 import com.powsybl.iidm.network.Bus;
 import com.powsybl.iidm.network.Generator;
 import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.ShuntCompensator;
 
+import java.util.Optional;
+import java.util.OptionalDouble;
 import java.util.stream.Stream;
 
 public final class RegulationUtils {
     private RegulationUtils() {
     }
 
-    public static Stream<Generator> getRegulatingGenerators(Network network, Bus bus) {
+    public static Stream<Generator> getRegulatingGenerators(Network network, Bus regulatingBus) {
         return network.getGeneratorStream().filter(Generator::isVoltageRegulatorOn)
-            .filter(g -> bus.equals(g.getRegulatingTerminal().getBusView().getBus()));
+            .filter(g -> regulatingBus.equals(g.getRegulatingTerminal().getBusView().getBus()));
+    }
+
+    public static Stream<ShuntCompensator> getRegulatingShuntCompensators(Network network, Bus regulatingBus) {
+        return network.getShuntCompensatorStream().filter(ShuntCompensator::isVoltageRegulatorOn)
+            .filter(shunt -> regulatingBus.equals(shunt.getRegulatingTerminal().getBusView().getBus()));
+    }
+
+    private static OptionalDouble getTargetVForRegulatingElement(Network network, Bus regulatingBus, String shuntId,
+                                                                 String generatorId) {
+        Optional<Double> optDouble = getRegulatingShuntCompensators(network, regulatingBus)
```

Comment:
I think we should add some comments to explain how this method is functioning:
```
// The voltage setpoint to use is the same as for the other regulating shunt compensators
// or generators regulating this bus
```
and before the last `or(...)`:
```
// If no other regulating element is found on the bus, use the network voltage
```